src/decawave/src/decawave_node.cpp

```diff
@@ -110,29 +85,28 @@ void update_callback(const ros::TimerEvent&){
   decawave::Range msg;
 
   // update decawave data
-  piTag->updateSamples();
-
-  // get tag position from the decawave
-  tagPos = piTag->getPos();
-
-  ros::param::get("~port_num",port_num);
-  std::string frame_id = std::to_string(port_num);
-
-  msg.distance = tagPos.x;
-  msg.estimated_variance = 0.10;
-  // fill out message header
-  msg.header.stamp = ros::Time::now();
-  msg.header.frame_id = "decawave" + frame_id;//"decawave_" + port_num;
-  msg.child_frame_id = "decawave2_link"; //change to correct part
-
-  gps_pub.publish(msg);
-
-  msg.distance = tagPos.y;
-  msg.estimated_variance = 0.10;
-  // fill out message header
-  msg.header.stamp = ros::Time::now();
-  msg.header.frame_id = "decawave" + frame_id;//"decawave_" + port_num;
-  msg.child_frame_id = "decawave3_link"; //change to correct part
-
-  gps_pub.publish(msg);
+  std::vector<anchor> anchors = piTag->updateSamples();
+
+  ros::param::get("~port_num",port_name);
+  std::string frame_id = port_name;
+
+  int j=anchors.size();
+  int i=0;
+  anchor m_anchor;
+  while (i<j){
+    m_anchor=anchors[i];//get new anchor
+    msg.distance = ((float)m_anchor.distance)/1000;
+    msg.distance_quality = m_anchor.distance_quality;
+    msg.quality_factor=m_anchor.quality_factor;
+    msg.position={((float)m_anchor.position[0])/1000,
+      ((float)m_anchor.position[1])/1000,
+      ((float)m_anchor.position[3])/1000};
+    // fill out message header
+    msg.header.stamp = ros::Time::now();
+    msg.header.frame_id = "decawave_" + frame_id;//"decawave_" + port_num;
+    msg.child_frame_id = "decawave2_link"; //change to correct part
```

Comment:
This shouldn't be necessary, and it's moreso confusing, because you only have 1 type of data, so it can only fit into 1 frame. 
(Contrast odometry data: the position could be in global-map frame, but the velocity and rotation might be in robot frame (ex fwd vel, spin rate))